src/components/CityColumnChart/CityColumnChart.js

```diff
@@ -73,26 +76,36 @@ const CityBarChart = (props) => {
 
   // series is an array even if there is only one indicator
   // so this works for a normal or a stacked chart
-  const series = props.indicatorIds.map((indicator, i) => ({
+  const series = props.indicatorIds.map((indicatorId, i) => ({
     index: props.indicatorIds.length - i, // reverse sort the series (to counteract Highcharts)
-    name: INDICATORS[indicator].name,
+    name: INDICATORS[indicatorId].name,
     color: getColorVariant(props.colorBase, SERIES_SHADES[i]),
-    data: getSeriesDataForIndicator(data, indicator),
+    data: getSeriesDataForIndicator(data, indicatorId),
   }));
 
   // we set a left margin on the chart explicitly
   // this means we can left align the title and the legend with the y axis
-  const marginLeft = 120;
+  const marginLeft = narrowViewport ? 15 : 120;
+
+  // We start losing visibility of x-axis labels at viewport widths lower than 550 
+  const xAxisLabelsConfig = narrowViewport
+    ? { rotation: -90, style: { fontSize: '9px' }, padding: 0 }
+    : { rotation: -45 };
```

Comment:
This should use the [Highcharts responsive feature](http://api.highcharts.com/highcharts/responsive). The code above won't update when a user rotates/resizes their device.